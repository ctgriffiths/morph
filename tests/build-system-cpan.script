#!/bin/sh
#
# Convert the hello-chunk project to perl with CPAN
#
# Copyright (C) 2011, 2012  Codethink Limited
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; version 2 of the License.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

set -eu

chunkrepo="$DATADIR/chunk-repo"
cd "$chunkrepo"

git checkout --quiet farrokh

git rm --quiet hello.c

cat <<EOF >hello
#!/usr/bin/perl
print "hello, world\n"
EOF
git add hello

cat <<EOF >Makefile.PL
use strict;
use warnings;
use ExtUtils::MakeMaker;
WriteMakefile(
    EXE_FILES => ['hello'],
)
EOF
git add Makefile.PL

cat <<EOF >hello.morph
{
    "name": "hello",
    "kind": "chunk",
    "build-system": "cpan"
}
EOF
git add hello.morph

git commit --quiet -m 'convert hello into a perl cpan project'

"$SRCDIR/scripts/test-morph" build --prefix=/ chunk-repo farrokh hello.morph

for chunk in "$DATADIR/cache/"*.chunk.*
do
    echo "$chunk:" | sed 's/[^.]*//'
    tar -tf "$chunk" | LC_ALL=C sort |
            sed -e '/^\.\/./s:^\./::' | grep -F "bin/hello"
done
