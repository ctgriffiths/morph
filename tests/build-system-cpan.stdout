.chunk.hello:
./
baserock/
baserock/hello.meta
bin/
bin/hello
lib/
lib/PERLDIR/
lib/PERLDIR/ARCH/
lib/PERLDIR/ARCH/auto/
lib/PERLDIR/ARCH/auto/hello.build/
lib/PERLDIR/ARCH/auto/hello.build/.packlist
lib/PERLDIR/ARCH/perllocal.pod
